pyluos/integration/freedomRobotics.py

```diff
@@ -0,0 +1,105 @@
+        # We have this service.
+        if hasattr(service, msg["message"]):
+            service_data = getattr(service, msg["message"])
+            service_data = msg["message"][msg["message"]]
+
+    def _update(self, alias, new_state):
+        if 'io_state' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/Joy", \
+                    {"io_state": new_state['io_state']})
+            # print ("/" + alias, \
+            #         "sensor_msgs/Joy", \
+            #         {"buttons": new_state['io_state']})
+        if 'temperature' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/Temperature", \
+                    {"temperature": new_state['temperature']})
+        if 'lux' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/Illuminance", \
+                    {"illuminance": new_state['lux']})
+        if 'rot_position' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/JointState", \
+                    {"position": new_state['rot_position'], "name":alias})
+        if 'trans_position' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/Range", \
+                    {"range": new_state['trans_position']})
+        if 'rot_speed' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/JointState", \
+                    {"velocity": new_state['rot_speed'], "name":alias})
+        if 'trans_speed' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/JointState", \
+                    {"velocity": new_state['trans_speed'], "name":alias})
+        if 'force' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/JointState", \
+                    {"effort": new_state['force'], "name":alias})
+        if 'current' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/BatteryState", \
+                    {"current": new_state['current']})
+        if 'volt' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/BatteryState", \
+                    {"voltage": new_state['volt']})
+        if 'quaternion' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/Imu", \
+                    {"orientation": {"y": new_state['quaternion'][0],"x": new_state['quaternion'][1],"z": new_state['quaternion'][2],"w": new_state['quaternion'][3]}})
+        if 'linear_accel' in new_state:
+            self._link.message("/" + alias, \
+                    "sensor_msgs/Imu", \
+                    {"linear_acceleration": {"y": new_state['linear_accel'][0],"x": new_state['linear_accel'][1],"z": new_state['linear_accel'][2]}})
+
+# I don't know what to do with those ones :
```

Comment:
Here's some useful lists of available messages for these:
* [geometry_msgs](https://docs.ros2.org/galactic/api/geometry_msgs/index-msg.html)
* [diagnostic_msgs](https://docs.ros2.org/galactic/api/diagnostic_msgs/msg/) good for health system information
* [nav_msgs](https://docs.ros2.org/galactic/api/nav_msgs/msg/) Odometry
* [pendulum_msgs](https://docs.ros2.org/galactic/api/pendulum_msgs/msg/JointState.html) for a single joint
* [sensor_msgs](https://docs.ros2.org/galactic/api/sensor_msgs/msg/) includes JointState message, for serial joint states
* [shape_msgs](https://docs.ros2.org/galactic/api/shape_msgs/msg/) for various shapes, useful for perception, computer vision
* [std_msgs](https://docs.ros2.org/galactic/api/std_msgs/msg/) for basic types
* [stereo_msgs](https://docs.ros2.org/galactic/api/stereo_msgs/msg/DisparityImage.html) for stereo cameras
* [UUID_msgs](https://docs.ros2.org/galactic/api/unique_identifier_msgs/msg/UUID.html)

The FR API also just accepts any string as a message type identifier along with a json blob (python dictionary when using `Link`, but we encourage using the ROS types because we have better visualization support for those. That doesn't matter if you're only using the API though.

Reply:
I will stick to the ROS message...
Thank you for the tips.